tiledb/sm/rest/curl.cc

```diff
@@ -486,17 +486,29 @@ void Curl::set_curl_request_options(
         "Cannot make curl request; curl instance is null.");
   }
   /* set url to fetch */
-  curl_easy_setopt(curl, CURLOPT_URL, url);
+  CURLcode rc = curl_easy_setopt(curl, CURLOPT_URL, url);
+  if (rc != CURLE_OK) {
+    throw CurlException("Error initializing libcurl; failed to set URL to fetch");
```

Comment:
Should we include libcurl's error message obtained with `curl_easy_strerror`?

Reply:
Done.